pkg/operator/apiserver/controller/workload/workload.go

```diff
@@ -353,6 +375,32 @@ func isUpdatingTooLong(operatorStatus *operatorv1.OperatorStatus, progressingCon
 	return progressing != nil && progressing.Status == operatorv1.ConditionTrue && time.Now().After(progressing.LastTransitionTime.Add(15*time.Minute)), nil
 }
 
+// hasDeploymentProgressed returns true if the deployment reports NewReplicaSetAvailable
+// via the DeploymentProgressing condition
+func hasDeploymentProgressed(status appsv1.DeploymentStatus) bool {
+	for _, cond := range status.Conditions {
+		if cond.Type == appsv1.DeploymentProgressing {
+			return cond.Status == corev1.ConditionTrue && cond.Reason == "NewReplicaSetAvailable"
+		}
+	}
+	return false
+}
+
+func countTerminatingPods(pods []*corev1.Pod) int32 {
+	numberOfTerminatingPods := 0
+	for _, p := range pods {
+		if isPodTerminating(p) {
+			numberOfTerminatingPods += 1
+		}
+	}
+	return int32(numberOfTerminatingPods)
+}
+
+func isPodTerminating(p *corev1.Pod) bool {
+	return p.Status.Phase != corev1.PodFailed && p.Status.Phase != corev1.PodSucceeded &&
+		p.DeletionTimestamp != nil
+}
```

Comment:
Just to make it clear, the pod is considered terminating when it has been marked for deletion but has not yet fully exited? If you happen to push again, could you add a comment for future reference?

Reply:
I have decided against the dependency on the terminating pods.